fastly/origin_inspector.go

```diff
@@ -0,0 +1,146 @@
+package fastly
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// OriginInspector represents the response format returned for a request to
+// the historical Origin Inspector metrics endpoint.
+type OriginInspector struct {
+	Data   []OriginData `mapstructure:"data"`
+	Meta   OriginMeta   `mapstructure:"meta"`
+	Status string       `mapstructure:"status"`
+}
+
+// OriginData represents the series of values over time for a single
+// dimension combination.
+type OriginData struct {
+	Dimensions map[string]string `mapstructure:"dimensions"`
+	Values     []OriginMetrics   `mapstructure:"values"`
+}
+
+// OriginMetrics represents the possible metrics that can be returned by a call
+// to the Origin Inspector endpoints.
+type OriginMetrics struct {
```

Comment:
General comment, I prefer struct fields to be sorted (unless memory allocation is a concern) as it makes it easier to search for fields when reviewing the code.

Reply:
That said, using a sort command over the latencies would produce an order that might be considered less than ideal...

```
Latency0to1ms         uint64 `mapstructure:"latency_0_to_1ms"`
Latency10000to60000ms uint64 `mapstructure:"latency_10000_to_60000ms"`
Latency1000to5000ms   uint64 `mapstructure:"latency_1000_to_5000ms"`
Latency100to250ms     uint64 `mapstructure:"latency_100_to_250ms"`
Latency10to50ms       uint64 `mapstructure:"latency_10_to_50ms"`
Latency1to5ms         uint64 `mapstructure:"latency_1_to_5ms"`
Latency250to500ms     uint64 `mapstructure:"latency_250_to_500ms"`
Latency5000to10000ms  uint64 `mapstructure:"latency_5000_to_10000ms"`
Latency500to1000ms    uint64 `mapstructure:"latency_500_to_1000ms"`
Latency50to100ms      uint64 `mapstructure:"latency_50_to_100ms"`
Latency5to10ms        uint64 `mapstructure:"latency_5_to_10ms"`
Latency60000ms        uint64 `mapstructure:"latency_60000ms"`
```